### DISCUSSION TITLE
HTTPError: 413 Client Error: Request Entity Too Large

### DISCUSSION THREAD

**Discussion by dpriskorn** [LeMyst/WikibaseIntegrator#462] [Q&A] [answered]
Does anyone know how to avoid this error?
https://github.com/internetarchive/wcdimportbot/issues/358
code running is https://github.com/internetarchive/wcdimportbot/commit/9b887558489e1016f15b3583b28fced8888b700c
Based on https://duckduckgo.com/?t=ffab&q=wikibase+Request+Entity+Too+Large+for+&ia=web and https://www.cyberciti.biz/faq/linux-unix-bsd-nginx-413-request-entity-too-large/ it seems that this is a configuration issue related to nginx at Wikibase.cloud?

See also https://phabricator.wikimedia.org/T324350

User story: as a developer I want to be able to limit the request size that WBI sends to Wikibase, so I can honor the request limits set in nginx.

See also https://github.com/internetarchive/wcdimportbot/issues/369 and related issues there.

**Comment by LeMyst** [comment, marked as answer]
You have these arguments that can help you in the `write()` method of each entity type (https://github.com/LeMyst/WikibaseIntegrator/blob/36f7acb09198253a25343bd14850d88039ae1508/wikibaseintegrator/entities/baseentity.py#L207):
* limit_claims: Limit to a list of specific claims to reduce the data sent and avoid sending the complete entity.
* field_to_update: A list or a single EntityField to update. If not set, all fields will be updated.

so you can do something like `write(data, limit_claims=['P1082'], fields_to_update=EntityField.CLAIMS)` to only update the Claims and more specifically the property P1082. All other fields or properties will not be updated.
